### DISCUSSION TITLE
Add CLI

### DISCUSSION THREAD

**Discussion by XTorLukas** [sunecz/Media-Downloader#7] [Ideas]
I would have a proposal for creating a CLI transformation.
Then it would be possible to create automated scripts.

**Comment by sunecz** [comment]
Hello, thank you for the proposal!

What kind of tasks would you find useful for the CLI variant? I suppose you would like to automate, for example, the downloading of the latest episode of a program? There are many details that would have to be specified, e.g. how to handle input from the user (URI, names, season/episode numbers, ...), what operations to support, etc.

For example, some simple ideas (`-d` for download, `-i` for information):
- `md -d --program "PROGRAM_NAME" --all`
  - Download all episodes of all seasons of the specified program.
- `md -d --program "PROGRAM_NAME" --season SEASON_NUMBER --all`
  - Download all episodes of the specified season.
  - For multiple seasons range format may be used: `--season START-END`.
- `md -d --program "PROGRAM_NAME" --season SEASON_NUMBER --episode EPISODE_NUMBER`
  - Download only the specified episode in the specified season.
  - For multiple episodes range format may be used: `--episode START-END`.
- `md -d --uri URI`
  - Download the specified URI.
- `md -i seasons --program "PROGRAM_NAME"`
  - Get seasons of the specified program.
- `md -i episodes --program "PROGRAM_NAME" --season SEASON_NUMBER`
  - Get episodes of the specified program and season.
- `md -i sources --program "PROGRAM_NAME" --season SEASON_NUMBER --episode EPISODE_NUMBER`
  - Get sources of the specified season and episode.
- `md -i sources --uri URI`
  - Get sources of the specified URI.

There are many operations that may be done. The ideas above don't even include the support for login credentials, additional configuration, title formats, etc. But I guess it may be a start.

However, in any case, CLI variant of the application is something that I would like to do.

Could you please elaborate on what would be useful to you? From that information we can move forward.

Thank you!
